contracts/stellar-interchain-token-service/src/contract.rs

```diff
@@ -696,13 +743,16 @@ impl InterchainTokenService {
         Ok(token_address)
     }
 
-    fn ensure_token_not_registered(env: &Env, token_id: BytesN<32>) -> Result<(), ContractError> {
+    fn ensure_token_not_registered(
```

Comment:
can this function be moved into token_id as well? The `new` function still allows the circumvention of the check (if it's a major pain to move I'm ok with leaving it)